### PR TITLE
including 12factor exception handling

### PR DESCRIPTION
**Pull request by jmervine**
After some debugging issues around timeout exceptions, I'm including more data in exception output for better splunking.

Note: This also adds more data to the rollbar exception as well.

@reidmix @lexelby @joshuatobin @stevo550

### CODE REVIEW

File: lib/umpire/log.rb [hunk 1]
```diff
@@ -3,14 +3,20 @@
...
       if Config.deploy == "test"
         blk.call if blk
       else
-        Scrolls.log({app: Config.app, deploy: Config.deploy}.merge(data), &blk)
+        Scrolls.log(BASE_DATA.merge(data), &blk)
       end
     end
 
+    def self.log_exception(ex, data = {})
+      Scrolls.log_exception(BASE_DATA.merge(data), ex)
```
Comment (reidmix) [inline lib/umpire/log.rb]: nice.  and `Scrolls.log_exception` will tie everything together with an `exception_id`

Comment (jmervine) [inline lib/umpire/log.rb]: Yeah, I noticed and <3 that.

File: lib/umpire/web.rb [hunk 2]
```diff
@@ -196,15 +203,25 @@ def create_aggregator(aggregation_method)
...
     error do
-      Rollbar.error(sinatra_err.message, class: sinatra_err.class, at: :internal_error, request_id: request_id)
-      log(at: "internal_error", "class" => sinatra_err.class, message: sinatra_err.message)
+      log_exception(sinatra_err)
       status 500
       JSON.dump({"error" => "internal server error", "request_id" => request_id}) + "\n"
     end
 
+    def self.base_log_data
+      { scope: Thread.current[:scope] }
```
Comment (reidmix) [inline lib/umpire/web.rb]: you might want to `freeze` it so it doesn't accidentally get merged with something.

**Comment by jmervine** [comment]
@reidmix addressed comment

File: lib/umpire/web.rb (hunk 2 above)
Comment (reidmix) [inline lib/umpire/web.rb]: actually you wound't need to freeze it since its being created everytime -- sorry about that.

**Comment by reidmix** [comment]
👍

**Pull request merged**